### PR TITLE
fix: data augmentation

### PR DESCRIPTION
**Pull request by streethagore**
Currently, the way data augmentation is set on the validation set erases the augmentation set for the training set because train_dataset and val_dataset are instances of a `Subset` class, which points to the same dataset: https://pytorch.org/docs/stable/_modules/torch/utils/data/dataset.html#Subset

Thus, when setting the augmentation for the validation set, it automatically sets the same augmentation for the training set.
Currently, as we do not apply augmentation on the validation set, this means that the augmentation was not applied to the training set.
Experiments should be ran again to look for direct improvements.

I decided to apply data augmentation on both train and validation set, for ease of implementation.
If a more fine-grained control over data augmentation is needed, I will need to update the pull request.

### CODE REVIEW

**Comment by sylvchev** [comment]
Ok with the patch to apply augmentation on both train and validation for now.

**Review by sylvchev — APPROVED** [review]

**Review by TheoRudkiewicz — COMMENTED** [review]
Thank you for the bug fix ! Definitely a well hidden one. Data augmentation on both training and validation seems the easier fix for now.
